appsmithorg/appsmith
# fix: 12225 default value behaviour in select and multi select widgets

## Description

Issue - the widgets render the provided default value even if it is unavailable in the options list.
Fix - Filter valid selections in the widget.

Fixes # (issue)

https://github.com/appsmithorg/appsmith/issues/12225

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

1. Create a new app. 
2. Drag in a select widget. The widget will render GREEN text as provided by the default value.
3. Change the options list to not include the option { value: "GREEN", label: "GREEN" }. Keep the default value as the same.
4. Refresh the widget. Observe that the widget renders with a default message : "Select option".
5. Repeat the same for multi select widget.

## Checklist:






## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>


    // Code coverage diff between base branch:release and head branch: fix-12225 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.16 **(-0.02)** | 37.65 **(-0.03)** | 35.65 **(-0.01)** | 56.4 **(-0.02)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**
 :red_circle: | app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx | 67.31 **(-13.15)** | 43.59 **(-13.08)** | 60 **(-10.59)** | 68.75 **(-11.74)**
 :red_circle: | app/client/src/widgets/SelectWidget/widget/index.tsx | 53.42 **(-1.51)** | 20.34 **(-1.88)** | 50 **(0)** | 54.29 **(-1.59)**</details>
